src/ucp/rma/flush.c

```diff
@@ -160,11 +160,14 @@ static void ucp_ep_flush_completion(uct_completion_t *self, ucs_status_t status)
 
     ucs_assert(!(req->flags & UCP_REQUEST_FLAG_COMPLETED));
 
+    req->status = status;
+
     if (status == UCS_OK) {
-        req->status = status;
+        ucp_ep_flush_progress(req);
+    } else {
+        req->send.state.uct_comp.count = 0;
```

Comment:
are we sure no one else will complete the request and make the counter drop below 0? IIRC the flush may be placed on more than one lane in the same time

Reply:
In case of error we stop flush progress then all other lanes will be destroyed by error handler

Reply:
can other lanes invoke the completion as well during their destruction?

Reply:
maybe we should just avoid calling ucp_ep_flush_progress() in case of error, and set request->send.flush.lanes to 0, so that already invoked flushes will complete, but new ones will not be invoked

Reply:
and also do req->send.state.uct_comp.count-- (instead of req->send.state.uct_comp.count=0)

Reply:
it will not work because uct iface ignores TX-completions on destroyed EPs

Reply:
> can other lanes invoke the completion as well during their destruction?

Did not check all TLs but should not

Reply:
so what happens if an operation is already posted to the transport - who calls the completion callback with status=ERR?

Reply:
it will be triggered only on first err completion of first failed lane, all others should be ignored.